fenixedu-ist-giaf-contracts/src/main/java/pt/ist/fenixedu/contracts/tasks/UpdateRetiredTeachersGroup.java

```diff
@@ -0,0 +1,59 @@
+package pt.ist.fenixedu.contracts.tasks;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.domain.User;
+import org.fenixedu.bennu.core.groups.DynamicGroup;
+import org.fenixedu.bennu.core.groups.Group;
+import org.fenixedu.bennu.core.groups.UserGroup;
+import org.fenixedu.bennu.scheduler.CronTask;
+import org.fenixedu.bennu.scheduler.annotation.Task;
+import org.joda.time.LocalDate;
+
+import pt.ist.fenixedu.contracts.domain.personnelSection.contracts.ContractSituation;
+import pt.ist.fenixedu.contracts.domain.personnelSection.contracts.GiafProfessionalData;
+import pt.ist.fenixedu.contracts.domain.personnelSection.contracts.PersonContractSituation;
+import pt.ist.fenixedu.contracts.domain.util.CategoryType;
+
+@Task(englishTitle = "Update Retired Teachers Group", readOnly = false)
+public class UpdateRetiredTeachersGroup extends CronTask {
+
+    @Override
+    public void runTask() throws Exception {
+        Set<ContractSituation> retirementSituations = Bennu.getInstance().getContractSituationsSet().stream()
+                .filter(cs -> cs.getName().getContent().matches(".*pose.*") || cs.getName().getContent().matches(".*eform.*")
```

Comment:
Doing something like this would be nicer to read:
`.filter(this::isRetiredSituation)`

With the following extracted method:

`    private boolean isRetiredSituation(final ContractSituation cs) {
    	final String name = cs.getName().getContent();
    	return name.matches(".*pose.*") || name.matches(".*eform.*") || name.matches(".*ubil.*") || name.matches(".*mérito.*");
	}`